src/components/editor-page/editor-page.tsx

```diff
@@ -80,6 +82,7 @@ export const EditorPage: React.FC = () => {
   useApplyDarkMode()
   useDocumentTitleWithNoteTitle()
   useEditorModeFromUrl()
+  useExtractFrontmatter(setFrontmatterInfo)
```

Comment:
tbh I don't know if I'm happy with this hook. This way the markdown content and the frontmatter information are separate and could be async. You change the markdown content and THEN you adjust the frontmatter based on the markdown. Also you need and editor-page-component. What about the read-only-page?
IMHO a better approach would be to calculate the frontmatter stuff in the redux reducer that sets the new markdown content. This way you make sure, that the markdown content and the note details in the redux state MUST be updated at the same time and are always in sync.
You could also store the cutted frontmatter raw text in the redux store so you know when you have to reparse it.

Reply:
My initial thought for this was that we could extract the frontmatter parsing into an asynchronously running web-worker later on. But as we discussed it a few days ago, this won't happen because the performance impact isn't worth the web-worker management.